pkg/lookup/lookup.go

```diff
@@ -82,6 +83,29 @@ func GetTemplates(
 	return templates
 }
 
+// GetMandatoryTemplates returns all ModuleTemplates for modules which are marked as mandatory with the given channel.
+func GetMandatoryTemplates(ctx context.Context, kymaClient client.Reader) (ModuleTemplatesByModuleName,
+	error,
+) {
+	moduleTemplateList := &v1beta2.ModuleTemplateList{}
+	if err := kymaClient.List(ctx, moduleTemplateList); err != nil {
+		return nil, fmt.Errorf("could not list ModuleTemplates: %w", err)
+	}
+
+	mandatoryModules := make(map[string]*ModuleTemplateTO)
+	for _, moduleTemplate := range moduleTemplateList.Items {
+		moduleTemplate := moduleTemplate
+		if moduleTemplate.Spec.Mandatory {
+			mandatoryModules[moduleTemplate.Name] = &ModuleTemplateTO{
+				ModuleTemplate: &moduleTemplate,
+				DesiredChannel: moduleTemplate.Spec.Channel,
```

Comment:
This `DesiredChannel` should not coming from moduletemplate, and even not assigned to a MandatoryTemplate, right? There is no channel required by MandatoryTemplate, we will probably make channel as optional field for `ModuleTemplate` in next API upgrade. To assign a DesiredChannel for MandatoryTemplate is confusion here.